ndk-build/src/apk.rs

```diff
@@ -176,27 +176,57 @@ impl Apk {
         }
     }
 
-    pub fn install(&self) -> Result<(), NdkError> {
+    pub fn install(&self, device_name: Option<String>) -> Result<(), NdkError> {
         let mut adb = self.ndk.platform_tool(bin!("adb"))?;
+
+        if let Some(device_name) = device_name {
+            adb.arg("-s").arg(device_name);
+        }
+
         adb.arg("install").arg("-r").arg(&self.path);
         if !adb.status()?.success() {
             return Err(NdkError::CmdFailed(adb));
         }
         Ok(())
     }
 
-    pub fn start(&self) -> Result<(), NdkError> {
-        let mut adb = self.ndk.platform_tool(bin!("adb"))?;
-        adb.arg("shell")
+    pub fn start(&self, device_name: Option<String>) -> Result<(), NdkError> {
+        let mut am_start = self.shell(&device_name)?;
+        am_start
             .arg("am")
             .arg("start")
+            .arg("-W")
```

Comment:
That's neat, I was wondering how this was implemented by @dvc94ch in `x`, turns out to be a sleep-loop: https://github.com/cloudpeers/xbuild/blob/e0ebb50bfa78cfad61931bb30c70a624b85fdddd/xbuild/src/devices/adb.rs#L175-L186

Reply:
It's probably semi-smart to keep a loop so that duplicate PIDs are not causing issues:

https://github.com/cloudpeers/xbuild/blob/e58cc09b4f2cb07c1fa0591f9d65cea435dd5646/xbuild/src/devices/adb.rs#L179

But IMO `am start` should be able to emit the `pid` instead of this hackery.